internals/garbage-collector.rst

```diff
@@ -107,6 +122,38 @@ isn't running at all!), and merging partitions, all with a small constant number
 With care, they also support iterating over a partition while objects are being added to - and
 removed from - it, which is frequently required while GC is running.
 
+GC for the free-threaded build
+------------------------------
+
+In the free-threaded build, Python objects contain a 1-byte field
+``ob_gc_bits`` that is used to track garbage collection related state. The
+field exists in all objects, including ones that do not support cyclic
+garbage collection.  The field is used to identify objects that are tracked
+by the collector, ensure that finalizers are called only once per object,
+and, during garbage collection, differentiate reachable vs. unreachable objects.
+
+.. code-block:: none
+
+    object -----> +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+ \
+                  |                     ob_tid                    | |
+                  +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+ |
+                  | pad | ob_mutex | ob_gc_bits |  ob_ref_local   | |
```

Comment:
Do each `--+` mark one bit?  If so, `ob_tid`, `ob_ref_shared` and `*ob_type` are 2 bytes (16 bits), but this fields don't seem to match the description, since `ob_gc_bits` takes ~4 bits and `ob_ref_local` 6.

Reply:
No -- each row is 64-bits (on 64-bit platforms), but otherwise the fields `pad`, `ob_mutex`, `ob_gc_bits`, and `ob_ref_local` are not to scale.

I'm not sure what to do about this. It'd be nice if the diagram were to scale, but I think that would require the diagram to be very wide -- `ob_gc_bits` is only 1 byte, but 10 characters.

Reply:
It might be enough to just add a note after the diagram saying that e.g. "Note that `pad`, `ob_mutex`, `ob_gc_bits`, and `ob_ref_local` are not to scale and take X, Y, Z, W bytes respectively."

Another option, if those 4 values are related, could be to use a single name (like "free-threaded info bytes") and then break that down afterwards, either in text or with a separate "zoomed in" diagram.